### PR TITLE
[FDE-2530] Add Qwen3-ASR fine-tuning cookbook

### PR DESCRIPTION
**Pull request by lpan-b10**
## Tracking

- Linear: [FDE-2530 — Create Qwen3 ASR fine tuning cookbook](https://linear.app/baseten/issue/FDE-2530/create-qwen3-asr-fine-tuning-cookbook)

## What changed

- add a runnable `examples/qwen3-asr-transformers` cookbook based on QwenLM's official Qwen3-ASR SFT recipe, with training sources under `training/` and deployment sources kept separately under `truss/`
- configure one H100 by default with 16 CPU cores and 96 GiB RAM, with optional 2/4/8-GPU single-node DDP
- mount `Qwen/Qwen3-ASR-1.7B`, persist cache/checkpoints, and pin a compatible CUDA/FlashAttention/Qwen-ASR environment
- add generic Hugging Face audio dataset preparation with 16 kHz WAV materialization, duration filtering, Qwen target formatting, and deterministic train/eval splitting
- keep Hugging Face authentication optional for the public model/dataset defaults, with an explicit `HF_TOKEN_SECRET_NAME` opt-in for gated or private datasets
- add an adapted full-parameter BF16 trainer with FlashAttention 2, optional gradient checkpointing, resume support, inferable periodic checkpoints, and a stable `final/` checkpoint
- add `torchrun` launch support driven by `BT_NUM_GPUS`, with gradient accumulation scaled to preserve the default global batch of 128
- add a `truss/` deployment for `rank-0/final`, aligned with the Qwen3-ASR model-registry vLLM preset, plus an OpenAI-compatible audio client
- document batch-size/VRAM behavior, dataset overrides, local smoke testing, checkpoint persistence, deployment, and inference

## Why

The upstream recipe is a raw JSONL trainer. This packages it as an end-to-end Baseten cookbook with sensible H100 defaults and reusable dataset preparation, following the existing Qwen3-TTS example.

## Validation

- Python compile check
- `bash -n examples/qwen3-asr-transformers/training/run.sh`
- Ruff check and format check
- standalone mypy check on the Python files
- Truss schema validation for the checkpoint-backed deployment
- OpenAI client import and ASR response parsing checks
- local preparation smoke test using `hf-internal-testing/librispeech_asr_dummy`
- GitHub Actions example validation: [workflow run `32082228352`](https://github.com/basetenlabs/ml-cookbook/actions/runs/32082228352), [training job `wlx7xvw`](https://app.baseten.co/training/w676e5q/logs/wlx7xvw)
  - discovered `examples/qwen3-asr-transformers/training/config.py` through the standard recipe layout
  - submitted the public default without a Hugging Face workspace secret and reached `TRAINING_JOB_COMPLETED`
  - persisted a 4.08 GB full `checkpoint-3` artifact and completed workflow cleanup
- Baseten H100 end-to-end smoke test in FDE Internal, team `lei`: [training job `qk7dllw`](https://app.baseten.co/training/qklvel3/logs/qk7dllw)
  - prepared 7 train / 1 eval examples
  - completed one BF16 optimizer step and evaluation
  - eval loss: 1.0948
  - wrote `/mnt/ckpts/final`
  - persisted a 4.08 GB full checkpoint
- Baseten 2xH100 DDP smoke test in FDE Internal, team `lei`: [training job `q9vxo63`](https://app.baseten.co/training/q95mzjw/logs/q9vxo63)
  - launched two Trainer ranks with `torchrun --nproc-per-node=2`
  - prepared 4 train / 2 eval examples and completed one synchronized optimizer step
  - train loss: 21.6748; eval loss: 22.8206
  - wrote `/mnt/ckpts/final` from rank 0
  - checkpoint sync completed with both `checkpoint-1` and `final`
- Baseten deployment smoke test in FDE Internal, team `lei`: [model `qjd4dklq`](https://app.baseten.co/models/qjd4dklq), [deployment `w7mzyr0`](https://app.baseten.co/models/qjd4dklq/logs/w7mzyr0)
  - pulled the synchronized 4.1 GB `rank-0/final` artifact from job `qk7dllw`
  - became active on one H100 with scale-to-zero enabled
  - returned HTTP 200 from `/v1/chat/completions` in 5.54 seconds
  - produced the expected English transcript through both curl and `truss/call.py`
  - no error-level deployment logs during the smoke test

### CODE REVIEW

**Review by iancarrasco-b10 — APPROVED** [review]

File: examples/qwen3-asr-transformers/training/run.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    python3 -m venv .venv
+fi
+
+source .venv/bin/activate
+python -m pip install -q -r requirements.txt
+
+# Dataset defaults: a bounded slice of LibriSpeech train-clean-100. Override
+# these for another Hugging Face audio dataset.
+DATASET_REPO="${DATASET_REPO:-openslr/librispeech_asr}"
+DATASET_CONFIG="${DATASET_CONFIG:-clean}"
```
[the remark this answers is not in the document]
Comment (lpan-b10) [reply examples/qwen3-asr-transformers/training/run.sh]: Fixed in db2c6f3: `DATASET_CONFIG` and `LANGUAGE` now use the non-colon default expansion, so their defaults apply only when unset and explicitly empty values are preserved. Verified both unset and empty cases locally, and the end-to-end PR training workflow passed: https://github.com/basetenlabs/ml-cookbook/actions/runs/32166310576

**Review by ShreyaRavi — APPROVED** [review]
very cool!
